mysql/README.md

```diff
@@ -1,41 +1,88 @@
-dockerfiles-fedora-MySQL
-========================
+MySQL for general usage and OpenShift - Docker image
+======================================================
 
-This repo contains a recipe for making Docker container for SSH and MySQL on Fedora. 
+This repository contains Dockerfiles for MySQL images for general usage and for OpenShift.
 
-Check your Docker version
 
-    # docker version
+Environment variables and volumes
+----------------------------------
 
-Perform the build
+The image recognizes the following environment variables that you can set during
+initialization by passing `-e VAR=VALUE` to the Docker run command.
 
-    # docker build --rm -t <yourname>/mysql .
+|    Variable name       |    Description                            |
+| :--------------------- | ----------------------------------------- |
+|  `MYSQL_USER`          | User name for MySQL account to be created |
+|  `MYSQL_PASSWORD`      | Password for the user account             |
+|  `MYSQL_DATABASE`      | Database name                             |
+|  `MYSQL_ROOT_PASSWORD` | Password for the root user (optional)     |
 
-Check the image out.
+The following environment variables influence the MySQL configuration file. They are all optional.
 
-    # docker images
+|    Variable name                |    Description                                                    |    Default
+| :------------------------------ | ----------------------------------------------------------------- | -------------------------------
+|  `MYSQL_LOWER_CASE_TABLE_NAMES` | Sets how the table names are stored and compared                  |  0
+|  `MYSQL_MAX_CONNECTIONS`        | The maximum permitted number of simultaneous client connections   |  151
+|  `MYSQL_FT_MIN_WORD_LEN`        | The minimum length of the word to be included in a FULLTEXT index |  4
+|  `MYSQL_FT_MAX_WORD_LEN`        | The maximum length of the word to be included in a FULLTEXT index |  20
+|  `MYSQL_AIO`                    | Controls the `innodb_use_native_aio` setting value in case the native AIO is broken. See http://help.directadmin.com/item.php?id=529 |  1
 
```

Comment:
So... I didn't particularly like this.  The interface isn't too bad, but I can't help thinking it's one of the reasons the image ends up being so much more complicated. (I saw that diffstat :-P).

This doesn't particularly sound like what you'd do for a "set of examples that can assist in learning about Docker" etc. My opinion - without having much experience - was that users should derive a Dockerfile with their desired config files.

There's an LWN post about the Kubernetes 1.2 release which makes a related point. It mentions a new ConfigMap feature which injects config files at runtime.  http://lwn.net/Articles/680833/

Maybe the AIO thing is important enough it needs to be kept?  In which case it would be nice to have an example scenario where it's needed.  I'm imagining "Some Docker hosting services are known to require this", or "Some container hosting services are known to require this" (euphemism for VPS) if we can't provide anything more specific.  I.e. you might end up needing it when deploying, but it shouldn't be needed on Fedora, right?